File: src/main/java/arcana/common/items/ArcanaGroup.java
package arcana.common.items;

import arcana.Arcana;
import arcana.common.items.ModItems;
import net.minecraft.item.Item;
import net.minecraft.item.ItemGroup;
import net.minecraft.item.ItemStack;

public class ArcanaGroup extends ItemGroup {
    public ArcanaGroup(){
        super(Arcana.id);
        this.hideTitle();
    }
    @Override public ItemStack makeIcon() {return new ItemStack(ModItems.FIREWAND);}
    @Override public boolean hasSearchBar() {return false;}

    public static Item.Properties itemProps(){
        return new Item.Properties().tab(Arcana.ARCANAGROUP);
    }
}

File: src/main/kotlin/arcana/common/items/ArcanaGroup.kt
package arcana.common.items

import arcana.Arcana
import net.minecraft.item.Item
import net.minecraft.item.ItemGroup
import net.minecraft.item.ItemStack

object ArcanaGroup : ItemGroup(Arcana.id) {
    init {
        hideTitle()
    }

    override fun makeIcon() = ItemStack(ModItems.FIREWAND)
    
    override fun hasSearchBar() = false

    val props: Item.Properties
        get() = Item.Properties().tab(ArcanaGroup)
}